nixos/modules/services/networking/go-away.nix

```
@@ -0,0 +1,257 @@
+  policyEtcPath = name: "go-away/${if name == "" then "default" else name}-policy.yaml";
+  policyFile =
+    name: instance:
+    pkgs.runCommandLocal "${instanceName name}-policy.yaml"
+      {
+        nativeBuildInputs = [ cfg.package ];
+      }
+      ''
+        cat ${yamlFormat.generate "${instanceName name}-policy.yaml" instance.policy} > $out
+        go-away -check -config ${configFile name instance} -policy $out
+      '';
+in
+{
+  options.services.go-away = {
+    package = lib.mkPackageOption pkgs "go-away" { };
+
+    instances = lib.mkOption {
+      default = { };
+      description = ''
+        An attribute set of Go-Away instances.
+
+        The attribute name may be an empty string, in which case the `-<name>` suffix is not added to the service name
+        and socket paths.
+      '';
+      type = types.attrsOf (
+        types.submodule (
+          { name, config, ... }:
+          {
+            options = {
+              enable = lib.mkEnableOption "this instance of Go-Away" // {
+                default = true;
+              };
+
+              extraFlags = lib.mkOption {
+                default = [ ];
+                description = "A list of extra flags to be passed to Go-Away.";
+                example = [ "-slog-level DEBUG" ];
+                type = types.listOf types.str;
+              };
+
+              bindHost = lib.mkOption {
+                default = "";
+                type = types.str;
+              };
+
+              bindPort = lib.mkOption {
+                type = types.ints.positive;
+              };
+
+              useACMEHost = lib.mkOption {
+                default = null;
+                type = types.nullOr types.str;
+              };
+
+              tlsCertificate = lib.mkOption {
+                default =
+                  if config.useACMEHost != null then
+                    "${hostConfig.security.acme.certs.${config.useACMEHost}.directory}/fullchain.pem"
+                  else
+                    null;
+                type = types.nullOr types.str;
+              };
+
+              tlsKey = lib.mkOption {
+                default =
+                  if config.useACMEHost != null then
+                    "${hostConfig.security.acme.certs.${config.useACMEHost}.directory}/key.pem"
+                  else
+                    null;
+                type = types.nullOr types.str;
+              };
+
+              config = lib.mkOption {
+                default = { };
+                description = ''
+                  TODO
+                '';
+                type = types.submodule {
+                  freeformType = yamlFormat.type;
+                  options = {
+                    bind = lib.mkOption {
+                      default = { };
+                      type = types.submodule {
+                        freeformType = yamlFormat.type;
+                        options = {
```

```suggestion
                    bind = {
```

I think that is nesting it a bit too much. You can make bind an attribute set just like that and define options below it. The yaml format is valid for everything in the settings option.